src/main/java/software/amazon/cloudwatchlogs/emf/model/MetricsContext.java

```diff
@@ -201,12 +199,38 @@ public MetricsContext createCopyWithContext() {
     }
 
     /**
-     * Serialize the metrics in this context to a string.
+     * Serialize the metrics in this context to strings.
      *
-     * @return the serialized string
+     * @return the serialized strings.
      * @throws JsonProcessingException if there's any object that cannot be serialized
      */
-    public String serialize() throws JsonProcessingException {
-        return this.rootNode.serialize();
+    public List<String> serialize() throws JsonProcessingException {
+        if (rootNode.metrics().size() <= Constants.MAX_METRICS_PER_EVENT) {
```

Comment:
I recommend adding a comment about what you’re doing here and why. Unfortunately, this bit of complexity exists because of constraints in the backend and it may not be clear to others what’s happening here.